massa-ledger-exports/src/key.rs

```diff
@@ -1,58 +1,77 @@
 use massa_models::{
-    address::{Address, AddressDeserializer, ADDRESS_SIZE_BYTES},
+    address::{Address, AddressDeserializer},
     serialization::{VecU8Deserializer, VecU8Serializer},
 };
-use massa_serialization::{DeserializeError, Deserializer, SerializeError, Serializer};
+use massa_serialization::{Deserializer, SerializeError, Serializer};
 use nom::error::{ContextError, ParseError};
 use std::ops::Bound::Included;
 
 pub const BALANCE_IDENT: u8 = 0u8;
 pub const BYTECODE_IDENT: u8 = 1u8;
 pub const DATASTORE_IDENT: u8 = 2u8;
 
-/// Balance key formatting macro
-#[macro_export]
-macro_rules! balance_key {
-    ($addr:expr) => {
-        [&$addr.prefixed_bytes()[..], &[BALANCE_IDENT]].concat()
-    };
+#[repr(u8)]
+#[derive(PartialEq, Eq, Clone, Debug)]
+pub enum KeyType {
+    BALANCE = 0,
+    BYTECODE = 1,
+    DATASTORE(Option<Vec<u8>>) = 2,
 }
 
-/// Bytecode key formatting macro
-///
-/// NOTE: still handle separate bytecode for now to avoid too many refactor at once
-#[macro_export]
-macro_rules! bytecode_key {
-    ($addr:expr) => {
-        [&$addr.prefixed_bytes()[..], &[BYTECODE_IDENT]].concat()
-    };
+impl KeyType {
```

Comment:
https://github.com/massalabs/massa/blob/589ae01ddc6ec19234e6cb726a8feab51acd4a43/massa-models/src/operation.rs#L234

Reply:
Oh, so I think what is interesting there isn't the `repr(u32)` but the use of `IntoPrimitive` and `TryFromPrimitive` which allows us to do serialization and deserialization in a cleaner way.

I will change it to match that way.

Reply:
It's not possible to use `IntoPrimitive` and `TryFromPrimitive` on the `KeyType` enum because of the DATASTORE key.

I don't see a better way of doing it than the one I used. If you have one, could you explain me ? If needed, we can make a call to talk about it.

Reply:
As an example, you can define:

`

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
pub enum OperationType {
    ...
}

#[derive(IntoPrimitive, Debug, Eq, PartialEq, TryFromPrimitive)]
#[repr(u32)]
enum OperationTypeId {
    ...
}
`

so the original enum does not change at all.

Reply:
Fixed with https://github.com/massalabs/massa/pull/3455/commits/1740e8c506318e805d7f0ee69045665ff52861e8